cg/apps/tb/models.py

```diff
@@ -1,49 +1,41 @@
-import datetime as dt
-from pathlib import Path
 from typing import Optional
 
-from dateutil.parser import parse as parse_datestr
-from pydantic.v1 import BaseModel, validator
+from pydantic import AfterValidator, BaseModel, ConfigDict, FieldValidationInfo, field_validator
+from typing_extensions import Annotated
+
+from cg.apps.tb.validators import parse_str_to_datetime, parse_str_to_path
 
 
 class TrailblazerAnalysis(BaseModel):
     id: int
     family: str
-    case_id = Optional[str]
-
-    version: Optional[str]
-    logged_at: Optional[str]
-    started_at: Optional[str]
-    completed_at: Optional[str]
-    status: Optional[str]
-    priority: Optional[str]
-    out_dir: Optional[str]
-    config_path: Optional[str]
-    comment: Optional[str]
-    is_deleted: Optional[bool]
-    is_visible: Optional[bool]
-    type: Optional[str]
-    user_id: Optional[int]
+    case_id: Optional[str] = None
+
+    @field_validator("case_id")
+    @classmethod
```

Comment:
Nice! Found out in a previous PR that if the ordering of the decorators are switched, then the validation never takes place. So this is the correct way!

Reply:
Good to know, was not aware it breaks